bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -705,12 +698,12 @@ public Bookie(ServerConfiguration conf, StatsLogger statsLogger, ByteBufAllocato
         stateManager = initializeStateManager();
         // register shutdown handler using trigger mode
         stateManager.setShutdownHandler(exitCode -> triggerBookieShutdown(exitCode));
-        // Initialise ledgerDirMonitor. This would look through all the
+        // Initialise dirsMonitor. This would look through all the
         // configured directories. When disk errors or all the ledger
         // directories are full, would throws exception and fail bookie startup.
-        this.ledgerMonitor = new LedgerDirsMonitor(conf, diskChecker, ledgerDirsManager);
+        this.dirsMonitor = new LedgerDirsMonitor(conf, diskChecker, Arrays.asList(ledgerDirsManager, indexDirsManager));
```

Comment:
if indexDirs==null, then indexDirsManager = ledgerDirsManager. So you should check this case before passing it to LedgerDirsMonitor. Otherwise LedgerDirsMonitor will check twice if  indexDirsManager == ledgerDirsManager

Reply:
Thanks @reddycharan , that's right.